src/coreComponents/physicsSolvers/fluidFlow/ReactiveCompositionalMultiphaseOBLKernels.hpp

```diff
@@ -0,0 +1,1427 @@
+   * @param[in] OBLOperatorsTable the OBL table function kernel
+   */
+  OBLOperatorsKernel( ObjectManagerBase & subRegion,
+                      MultivariableTableFunctionStaticKernel< numDofs, numOps > OBLOperatorsTable )
+    :
+    m_OBLOperatorsTable( OBLOperatorsTable ),
+    m_pressure( subRegion.getExtrinsicData< extrinsicMeshData::flow::pressure >() ),
+    m_compFrac( subRegion.getExtrinsicData< extrinsicMeshData::flow::globalCompFraction >() ),
+    m_temperature( subRegion.getExtrinsicData< extrinsicMeshData::flow::temperature >() ),
+    m_dPressure( subRegion.getExtrinsicData< extrinsicMeshData::flow::deltaPressure >() ),
+    m_dCompFrac( subRegion.getExtrinsicData< extrinsicMeshData::flow::deltaGlobalCompFraction >() ),
+    m_dTemperature( subRegion.getExtrinsicData< extrinsicMeshData::flow::deltaTemperature >() ),
+    m_OBLOperatorValues ( subRegion.getExtrinsicData< extrinsicMeshData::flow::OBLOperatorValues >()),
+    m_OBLOperatorDerivatives ( subRegion.getExtrinsicData< extrinsicMeshData::flow::OBLOperatorDerivatives >())
+  {}
+
+  /**
+   * @brief Compute the operator values and derivatives for an element
+   * @param[in] ei the element index
+   */
+  GEOSX_HOST_DEVICE
+  void compute( localIndex const ei ) const
+  {
+    arraySlice1d< real64 const, compflow::USD_COMP - 1 > const compFrac = m_compFrac[ei];
+    arraySlice1d< real64 const, compflow::USD_COMP - 1 > const dCompFrac = m_dCompFrac[ei];
+    arraySlice1d< real64, compflow::USD_OBL_VAL - 1 > const & OBLVals = m_OBLOperatorValues[ei];
+    arraySlice2d< real64, compflow::USD_OBL_DER - 1 > const & OBLDers = m_OBLOperatorDerivatives[ei];
+    real64 state[numDofs];
+
+    // we need to convert pressure from Pa (internal unit in GEOSX) to bar (internal unit in DARTS)
+    state[0] = (m_pressure[ei] + m_dPressure[ei]) * pascalToBarMult;
+
+    for( integer i = 1; i < numComps; ++i )
+    {
+      state[i] = compFrac[i - 1] + dCompFrac[i - 1];
+    }
+
+    if( enableEnergyBalance )
+    {
+      state[numDofs - 1] = (m_temperature[ei] + m_dTemperature[ei]);
+    }
+
+    m_OBLOperatorsTable.compute( state, OBLVals, OBLDers );
+
+    // we do not perform derivatives unit conversion here:
+    // instead we postpone it till all the derivatives are fully formed, and only then apply the factor only once in 'complete' function
+    // scaling the whole system might be even better solution (every pressure column needs to be multiplied by pascalToBarMult)
+  }
+
+protected:
+
+  // inputs
+  MultivariableTableFunctionStaticKernel< numDofs, numOps > m_OBLOperatorsTable;
+
+  // Views on primary variables and their updates
+  arrayView1d< real64 const > m_pressure;
+  arrayView2d< real64 const, compflow::USD_COMP > m_compFrac;
+  arrayView1d< real64 const > m_temperature;
+
+  arrayView1d< real64 const > m_dPressure;
+  arrayView2d< real64 const, compflow::USD_COMP > m_dCompFrac;
+  arrayView1d< real64 const > m_dTemperature;
```

Comment:
@francoishamon couldn't we use some "multidim" containers?

Reply:
I implemented the (field_n,field) formulation in the OBL solver in the last commit, so now `m_dPressure`, `m_dCompFrac`, and `m_dTemperature` are gone and the code is lighter there. 

But still I agree with you that we still pass many vectors to the compositional kernels which will consume a lot of registers on the GPU (I am really not expert in that). Maybe when we start optimizing more the compositional kernels, we can start merging the arrays to save some register space on GPU.

In this respect, the OBL PR is interesting, because all the properties are bundled in two vectors (`OBLOperatorValues` and `OBLOperatorValues_n` + derivatives) sized using "physics-based" templates at the beginning of the simulation.